server/game/AbilityTarget.js

```diff
@@ -139,7 +139,7 @@ class AbilityTarget {
                 return true;
             },
             onCancel: () => {
-                if(this.ifAble) {
+                if(this.selector.rejectAllowed(context)) {
```

Comment:
I'm not sure it's a good idea to push down even more targeting game logic into selectors, which were meant to be more about generic card selection prompts. The current implementation is already fuzzy, this breaks it down even further.

Reply:
@ystros Fair enough, but there were adjustments required for "A Traitor's Whisper" to properly function. This adjustment (as well as adding `rejectAllowed` to `BaseCardSelector.js` and `EachPlayerCardSelector.js`) is there to ensure players can only "cancel" the selection if they have no selections they can make; otherwise they are forced to.

Any alternate suggestions on how to do this? Otherwise, we will need to keep `rejectAllowed`, at least for now.

Reply:
Ah, so the problem is the `if able` applies per player, rather than the overall selection (e.g. if you can select for one player, you must do so)?

That probably means that the very existence of `EachPlayerCardSelector` doesn't match where I think we should be headed towards. Which means changing it to something else will be a much larger project.

Resolving for now.